api/db/thread.js

```diff
@@ -0,0 +1,12 @@
+import mongoose from 'mongoose';
+
+const threadSchema = new mongoose.Schema({
+  name: String,
+  isRead: Boolean,
+  messages: []
```

Comment:
I'm not too sure which design is preferable (nested vs flat+ref). But from experience I tend to think it's better to decouple them. aka just have a threadId on the message instead

I'm happy to discuss if you disagree :+1: